pkg/controllers/datastoresyncer/datastoresyncer.go

```diff
@@ -178,7 +193,43 @@ func (d *DatastoreSyncer) ensureExclusiveEndpoint(syncer *broker.Syncer) error {
 	return nil
 }
 
-func (d *DatastoreSyncer) createLocalCluster(federator federate.Federator) error {
+func (d *DatastoreSyncer) startNodeWatcher(stopCh <-chan struct{}) error {
+	nodeName, ok := os.LookupEnv("NODE_NAME")
+	if !ok {
+		// Healthcheck in globalnet deployments will not work because of missing NODE_NAME.
+		klog.Error("Error reading the NODE_NAME from the env, healthChecker functionality will not work.")
+	} else {
+		d.localNodeName = nodeName
+		resourceWatcher, err := watcher.New(&watcher.Config{
+			Scheme:     scheme.Scheme,
+			RestConfig: d.syncerConfig.LocalRestConfig,
+			ResourceConfigs: []watcher.ResourceConfig{
+				{
+					Name:                "Node watcher for datastoresyncer",
+					ResourceType:        &k8sv1.Node{},
+					ResourcesEquivalent: d.isNodeEquivalent,
+					Handler: watcher.EventHandlerFuncs{
+						OnCreateFunc: d.handleCreateOrUpdateNode,
+						OnUpdateFunc: d.handleCreateOrUpdateNode,
+						OnDeleteFunc: nil,
+					},
+				},
+			},
+		})
```

Comment:
super-nit: I would separate this to another function "createNodeWatcher" , so the if branches are easier to follow.

Reply:
Done